docker/scripts/startup.sh

```diff
@@ -1,23 +1,84 @@
-#!/bin/sh
+#!/bin/bash
 
-while getopts f:s:t:e:w: flag
+# Defaults
+DEBUG_MODE="false"
+
+while getopts d:f:s:t:e:w:b:m: flag
 do
     case "${flag}" in
+        d) DATASET=${OPTARG};;
         f) FRAMEWORK=${OPTARG};;
         s) SUBMISSION_PATH=${OPTARG};;
         t) TUNING_SEARCH_SPACE=${OPTARG};;
         e) EXPERIMENT_NAME=${OPTARG};;
         w) WORKLOAD=${OPTARG};;
+        b) DEBUG_MODE=${OPTARG};;
+        m) MAX_STEPS=${OPTARG};;
     esac
 done
 
-echo "Checking GPU presence and CUDA linking"
+ROOT_DATA_BUCKET="gs://mlcommons-data"
+ROOT_DATA_DIR="/data"
+
+EXPERIMENT_BUCKET="gs://mlcommons-runs"
+EXPERIMENT_DIR="/experiment_runs"
+
+if [ "${FRAMEWORK}" == "jax" ]
+then
+    COMMAND_PREFIX="python3"
+else 
+    COMMAND_PREFIX="torchrun --redirects 1:0,2:0,3:0,4:0,5:0,6:0,7:0 --standalone --nnodes=1 --nproc_per_node=8"
+fi
+
+if [ "${DATASET}" == "imagenet" ]
+then 
+    DATA_DIR="${ROOT_DATA_DIR}/${DATASET}/${FRAMEWORK}"
+    DATA_BUCKET="${ROOT_DATA_BUCKET}/${DATASET}/${FRAMEWORK}"
+else
+    DATA_DIR="${ROOT_DATA_DIR}/${DATASET}"
+    DATA_BUCKET="${ROOT_DATA_BUCKET}/${DATASET}"
+fi
+
+# Copy data from MLCommons bucket if data has not been downloaded yet
```

Comment:
it doesn't have to be now, but can you open an issue to do better checking for the existing data? I could definitely see for the larger datasets when we move the data around 100-1000s of times we have failures where it only partially copies it.

ideally we would do a check of the total number of bytes in the dir and/or a hash of the files, maybe with the option to disable in case people don't want it?

Reply:
Yes that's a good idea.